src/EnergyPlus/UnitarySystem.cc

```diff
@@ -8330,14 +8330,14 @@ namespace UnitarySystems {
                                       SupHeaterLoad,
                                       CompressorONFlag);
         Real64 CpAir = Psychrometrics::PsyCpAirFnW(state.dataLoopNodes->Node(this->CoolCoilInletNodeNum).HumRat);
+        Real64 heatCoildT =
+            (this->m_HeatCoilExists)
+                ? (state.dataLoopNodes->Node(this->HeatCoilOutletNodeNum).Temp - state.dataLoopNodes->Node(this->HeatCoilInletNodeNum).Temp)
+                : 0.0;
         Real64 CoolingOnlySensibleOutput =
             state.dataLoopNodes->Node(this->CoolCoilInletNodeNum).MassFlowRate * CpAir *
-            (state.dataLoopNodes->Node(this->NodeNumOfControlledZone).Temp - state.dataLoopNodes->Node(this->CoolCoilOutletNodeNum).Temp);
-        if (this->m_HeatCoilExists) {
-            CoolingOnlySensibleOutput -=
-                state.dataLoopNodes->Node(this->CoolCoilInletNodeNum).MassFlowRate * CpAir *
-                (state.dataLoopNodes->Node(this->HeatCoilOutletNodeNum).Temp - state.dataLoopNodes->Node(this->HeatCoilInletNodeNum).Temp);
-        }
+            ((state.dataLoopNodes->Node(this->NodeNumOfControlledZone).Temp - state.dataLoopNodes->Node(this->CoolCoilOutletNodeNum).Temp) -
+             heatCoildT);
```

Comment:
This issue was already corrected here previously. The change here updates the method to avoid a crash. If heating coil does not exist, HeatCoilOutletNodeNum and HeatCoilInletNodeNum = 0 and an array access exception occurs.